gix-object/src/tree/mod.rs

```diff
@@ -249,14 +249,57 @@ impl Ord for EntryRef<'_> {
     }
 }
 
+/// TODO
+/// Keep in mind that the path separator always is `/`, independent of the platform.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct RepositoryPathPuf {
+    /// TODO
+    inner: BString,
+}
+
+impl std::fmt::Display for RepositoryPathPuf {
+    #[inline]
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.inner.fmt(f)
+    }
+}
+
+impl From<&str> for RepositoryPathPuf {
+    fn from(value: &str) -> Self {
+        Self { inner: value.into() }
+    }
+}
+
+impl From<&BStr> for RepositoryPathPuf {
+    fn from(value: &BStr) -> Self {
+        Self { inner: value.into() }
+    }
+}
+
+impl From<BString> for RepositoryPathPuf {
```

Comment:
There should be no API that converts to this type without checking the input - paths now have to be relative.